app/Http/Controllers/ReportController.php

```diff
@@ -46,31 +48,39 @@ public function stock()
 
     		// We need to pull out all the records and insert them into array for processing
     		foreach($product->inbounds_with_lots as $inbound){
-                if($inbound->inbound->process_status != 'canceled') {
-        			$details->push(
-        				$this->formatStockDetails(
-    						$inbound->updated_at, 
-    						$inbound->lots->sum('pivot.quantity_received'), 
-    						0, 
-    						"Inbound - " . $inbound->inbound->display_no, 
-    						0
-    					)
-        			);
-                }
+				if($inbound->inbound->branch_id) {
+					if($inbound->inbound->branch_id == request()->selectedBranch){
```

Comment:
can merge the if
if($inbound->inbound->branch_id && $inbound->inbound->branch_id = =request()->selectedBranch)

But there is another better way which is to filter the inbounds_with_lots with only the selectedBranch to start, can reduce the number of records the system needs to loop